main/views.py

```diff
@@ -150,6 +243,22 @@ class SelfAssessPageView(LoginRequiredMixin, FormView[UserSkillsForm]):
     form_class = UserSkillsForm
     success_url = reverse_lazy("self-assess")
 
+    def get_context_data(self, **kwargs: Mapping[str, object]) -> dict[str, object]:
+        """Add the competencies framework data to the template context."""
+        context = super().get_context_data(**kwargs)
+        json_path = Path("data/skills-competencies-framework.json")
+        with open(json_path) as f:
+            framework = json.load(f)
+
+        for category in framework.get("categories", []):
+            category["competency_count"] = len(category.get("subcategories", []))
+            category["skill_count"] = sum(
+                len(sub.get("skills", [])) for sub in category.get("subcategories", [])
+            )
+
+        context["framework"] = framework
+        return context
+
```

Comment:
This is not used or needed in this `SelfAssessPageView`, the framework is taken directly from the database. I deleted this locally and it caused no problems.

```suggestion
```